pyogrio/_io.pyx

```diff
@@ -480,6 +480,22 @@ cdef detect_encoding(OGRDataSourceH ogr_dataset, OGRLayerH ogr_layer):
         return 'UTF-8'
 
     driver = get_driver(ogr_dataset)
+    return get_encoding_for_driver(driver)
+
+
+cdef get_encoding_for_driver(str driver):
+    """Get the typical encoding for the driver, or None.
+
+    Parameters
+    ----------
+    driver : str
+        driver to get the encoding for
+    
+    Returns
+    -------
+    str or None
+        the typical encoding if this is applicable for the driver, or None
+    """
     if driver == 'ESRI Shapefile':
         return 'ISO-8859-1'
```

Comment:
This should probably be set to `UTF-8` with a comment indicating that we are intentionally always setting `UTF-8` rather than the legacy value of `ISO-8859-1`.

Reply:
At least for shapefiles, this code is now only reached when reading... and I think it is safer to keep the default for reading to ISO-... rather than UTF-8?